### PR TITLE
Implement IGNORE NULLS for LAST_VALUE

### PR DESCRIPTION
**Pull request by huaxingao**
## Which issue does this PR close?

Closes #.

## Rationale for this change

Related https://github.com/apache/arrow-datafusion/pull/9411 which implemented `ignore null` in `First_value`.

This PR implements `ignore null` in `Last_value`

### CODE REVIEW

File: datafusion/physical-expr/src/aggregate/first_last.rs [hunk 1]
```diff
@@ -611,8 +634,20 @@ impl LastValueAccumulator {
...
+            // If ignoring nulls, find the last non-null value.
+            for index in indices.iter().flatten() {
+                if !value.is_null(index as usize) {
+                    return Ok(Some(index as usize));
+                }
+            }
+            Ok(None)
+        } else {
+            let indices = lexsort_to_indices(&sort_columns, Some(1))?;
+            Ok((!indices.is_empty()).then_some(indices.value(0) as _))
```
Comment (jayzhan211) [inline datafusion/physical-expr/src/aggregate/first_last.rs:649]: Why we get the first index here, but not the last one 😕

File: datafusion/core/tests/sql/aggregates.rs [hunk 2]
```diff
@@ -401,3 +401,83 @@ async fn test_first_value_with_sort() -> Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn last_value() -> Result<()> {
+    let session_ctx = SessionContext::new();
+    session_ctx
+        .sql("CREATE TABLE abc AS VALUES (1, 2, 3), (null, 5, 6)")
```
Comment (jayzhan211) [inline datafusion/core/tests/sql/aggregates.rs:409]: I think this kind of test are easier to maintain in `datafusion/sqllogictest/test_files/aggregate.slt`

File: datafusion/physical-expr/src/aggregate/first_last.rs (hunk 1 above)
Comment (mustafasrepo) [reply datafusion/physical-expr/src/aggregate/first_last.rs:649]: For `last_value` sort expressions are reversed. By this trick, we can get the first expression in the sort to get last one.

File: datafusion/sqllogictest/test_files/aggregate.slt [hunk 3]
```diff
@@ -3376,3 +3376,55 @@ SELECT FIRST_VALUE(column1 ORDER BY column2) IGNORE NULLS FROM t;
...
+
+query I
+SELECT LAST_VALUE(column1) IGNORE NULLS FROM t;
+----
+4
+
+statement ok
+DROP TABLE t;
+
+# Test for ignore null with ORDER BY in LASR_VALUE
```
Comment (mustafasrepo) [inline datafusion/sqllogictest/test_files/aggregate.slt:3402]: Not import. However there is a typo I guess.
```suggestion
# Test for ignore null with ORDER BY in LAST_VALUE
```

File: datafusion/sqllogictest/test_files/aggregate.slt [hunk 4]
```diff
@@ -3376,3 +3376,55 @@ SELECT FIRST_VALUE(column1 ORDER BY column2) IGNORE NULLS FROM t;
...
+NULL
+NULL
+
+query I
+SELECT LAST_VALUE(column1 ORDER BY column2 DESC) FROM t;
+----
+NULL
+
+query I
+SELECT LAST_VALUE(column1 ORDER BY column2 DESC) FROM t;
```
Comment (mustafasrepo) [inline datafusion/sqllogictest/test_files/aggregate.slt:3420]: This test an test above are exactly same. I think your original intent was to use `RESPECT NULLS` in one of them

**Review by mustafasrepo — APPROVED** [review]
LGTM!. Left minor comments. Also I agree with @jayzhan211. `SQL` tests should generally be in the `.slt` files for maintainability (unless desired behaviour cannot be tested from a test in the `.slt` file. Which should be very rare.).

File: datafusion/core/tests/sql/aggregates.rs (hunk 2 above)
Comment (huaxingao) [reply datafusion/core/tests/sql/aggregates.rs]: Removed. I have also removed `first_value` test I added last time

File: datafusion/sqllogictest/test_files/aggregate.slt (hunk 3 above)
Comment (huaxingao) [reply datafusion/sqllogictest/test_files/aggregate.slt]: Fixed. Thanks

File: datafusion/sqllogictest/test_files/aggregate.slt (hunk 4 above)
Comment (huaxingao) [reply datafusion/sqllogictest/test_files/aggregate.slt]: Fixed. Thanks

**Comment by mustafasrepo** [comment]
It seems like all reviews are addressed. Thanks @huaxingao for this PR.

**Comment by huaxingao** [comment]
Thanks @mustafasrepo @comphead @comphead

**Pull request merged**